packages/circuit-ui/components/List/List.tsx

```diff
@@ -81,24 +79,22 @@ const sizeStyles = ({ theme, size = 'one' }: ListProps & StyleProps) => {
   `;
 };
 
-const marginStyles = ({ noMargin }: ListProps) => {
+const marginStyles = ({ theme, noMargin }: StyleProps & ListProps) => {
   if (!noMargin) {
     if (
+      process.env.UNSAFE_DISABLE_NO_MARGIN_ERRORS !== 'true' &&
       process.env.NODE_ENV !== 'production' &&
       process.env.NODE_ENV !== 'test'
     ) {
-      deprecate(
-        'List',
-        'The default outer spacing in the List component is deprecated.',
-        'Use the `noMargin` prop to silence this warning.',
-        'Read more at https://github.com/sumup-oss/circuit-ui/issues/534.',
+      throw new Error(
+        'The List component requires the `noMargin` prop to be passed. Read more at https://github.com/sumup-oss/circuit-ui/issues/534.',
       );
     }
-
-    return null;
+    return css`
+      margin-bottom: ${theme.spacings.mega};
+    `;
   }
   return css`
-    margin-bottom: 0;
     li:last-child {
       margin-bottom: 0;
     }
```

Comment:
These styles also need to be migrated.

Reply:
Updated, can you check and let me know if you're happy with the changes. Thanks :)